runtime/caml/asm.h

```diff
@@ -45,6 +45,17 @@
 #define CFI_STARTPROC
 #endif
 
+/* .size and .type directives, with explicit end-of-directive semi-colon */
+#ifdef ASM_SIZE_TYPE_DIRECTIVES
+#define SIZE_DIRECTIVE(name) \
+        .size name, . - name;
+#define TYPE_DIRECTIVE(name,ty) \
+        .type name, ty;
+#else
+#define SIZE_DIRECTIVE(name)
+#define TYPE_DIRECTIVE(name,ty)
+#endif
+
```

Comment:
Similarly here we could have a `ASM_NONEXECSTACK_NOTE` that is defined conditionally, to avoid the conditional at use-sites.

Reply:
I just waited for an explicit suggestion to do so also :-)

Reply:
Done.